pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,325 @@
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (pr *policyRecommendationComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeAny
+}
+
+func (pr *policyRecommendationComponent) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		CreateNamespace(PolicyRecommendationNamespace, pr.cfg.Installation.KubernetesProvider, PSSRestricted),
+		allowTigeraPolicyForPolicyRecommendation(pr.cfg),
+	}
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.PullSecrets...)...)...)
+
+	objs = append(objs,
+		pr.serviceAccount(),
+		pr.clusterRole(),
+		pr.clusterRoleBinding(),
+		pr.deployment(),
+	)
+
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.ESSecrets...)...)...)
+
+	return objs, nil
+}
+
+func (pr *policyRecommendationComponent) Ready() bool {
+	return true
+}
+
+func (pr *policyRecommendationComponent) clusterRole() client.Object {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{"licensekeys", "managedclusters"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"crd.projectcalico.org"},
+			Resources: []string{"licensekeys"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{
+				"tiers",
+				"policyrecommendationscope",
+				"policyrecommendationscope/status",
+				"policyrecommendationscopes",
+				"policyrecommendationscopes/status",
+				"stagednetworkpolicies",
+				"tier.stagednetworkpolicies",
+				"stagednetworkpolicy",
+				"tier.stagednetworkpolicy",
+			},
+			Verbs: []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+		},
+	}
+
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PolicyRecommendationClusterRoleName,
+		},
+		Rules: rules,
+	}
+}
+
+func (pr *policyRecommendationComponent) clusterRoleBinding() client.Object {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PolicyRecommendationClusterRoleBindingName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     PolicyRecommendationClusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      PolicyRecommendationServiceAccountName,
+				Namespace: PolicyRecommendationNamespace,
+			},
+		},
+	}
+}
+
+func (pr *policyRecommendationComponent) controllerContainer() corev1.Container {
+	envs := []corev1.EnvVar{
+		{
+			Name:  "CLUSTER_NAME",
```

Comment:
The controller is going to be a managed cluster component? Or a management cluster component? Or both?

Does this config apply to both scenarios? If not, we should be checking the `managementClusterConfiguration` API to see which mode we're in and configuring appropriately.

Reply:
The deployment and subsequently the controller container is part of the management cluster component only. The check in policyrecommendation.go::Objects() checks:
```
// Deployment is for standalone or management cluster
	if !pr.cfg.ManagedCluster {
		objs = append(objs, pr.deployment())
	}
```

Currently, in the management cluster we'll find:
```
          - name: CLUSTER_NAME
            value: cluster
```

Should I remove this as unnecessarily defined?